core/domain/collection_services.py

```diff
@@ -526,6 +531,14 @@ def _save_collection(committer_id, collection, commit_message, change_list):
     else:
         collection.validate(strict=False)
 
+    for collection_node in collection.nodes:
+        if not exp_services.get_exploration_by_id(
+            collection_node.exploration_id, strict=False):
+            raise Exception(
```

Comment:
Well, the change list isn't part of the collection itself. I think the idea here is if a collection contains references to non-existent explorations, that's a flaw with the collection itself. This check was originally in Collection's validate function, but we removed it because it made that validation a bit too heavy. I think it still counts as a 'validation error,' though only when saving an exploration.